### DISCUSSION TITLE
Need help with Typings

### DISCUSSION THREAD

**Discussion by ispyinternet** [svecosystem/formsnap#171] [Q&A]
If anyone can tell me how to fix this I would be very grateful, thanks.

```
export let form: SuperForm<T>;
export let name: FormPath<T, boolean>
```

<img width="779" alt="Screenshot 2024-04-30 at 10 40 31" src="https://github.com/svecosystem/formsnap/assets/1741823/ad50e97c-e17a-4e53-b652-76730e01abf1">

**Comment by ambergristle** [comment]
formsnap is built on superforms, so this is a good place to start RE typing + generics:
- https://superforms.rocks/components#type-narrowing-for-paths

based on the error, it looks like T isn't sufficiently narrowed (e.g., `T extends Record<string, unknown>`). it may also be helpful to use `FormPathLeaves` instead of `FormPath`. i believe the former necessarily defines a path to a specific field/value, while the latter may terminate at an intermediate node (e.g., an object with nested fields)
